src/components/MapView/Analyser/index.tsx

```diff
@@ -187,8 +192,55 @@ function Analyser({ classes }: AnalyserProps) {
     };
 
     await dispatch(requestAndStoreAnalysis(params));
+    // eslint-disable-next-line fp/no-mutating-methods
+    history.push(
+      `?hazardLayerId=${hazardLayerId}&baselineLayerId=${baselineLayerId}&date=${selectedDate}&statistic=${statistic}&aboveThreshold=${aboveThreshold}&belowThreshold=${belowThreshold}`,
+    );
   };
 
+  useEffect(() => {
+    const queryString = history.location.search;
+    const params = new URLSearchParams(queryString);
+    const hazardLayerParamId = params.get('hazardLayerId');
+    const baselineLayerParamId = params.get('baselineLayerId');
+    const selectedParamDate: number = Number(params.get('date'));
+    const statisticParam = params.get('statistic');
+    const aboveThresholdParam: string = params.get('aboveThreshold') || '';
+    const belowThresholdParam: string = params.get('belowThreshold') || '';
+
+    if (!hazardLayerParamId || !baselineLayerParamId || !statisticParam) {
+      return;
+    }
+    function isHazardLayerId(layer: string): layer is LayerKey {
+      return typeof layer === 'string';
+    }
+    function isBaselineLAyerId(layer: string): layer is LayerKey {
+      return typeof layer === 'string';
+    }
+
+    if (isHazardLayerId(hazardLayerParamId)) {
+      setHazardLayerId(hazardLayerParamId);
+    }
+    if (isBaselineLAyerId(baselineLayerParamId)) {
+      setBaselineLayerId(baselineLayerParamId);
+    }
+
+    // Mean is selected by default
+    if (statisticParam === 'median') {
```

Comment:
Only mean is supported in the url?

Reply:
```
const [statistic, setStatistic] = useState(AggregationOperations.Mean);
```
This means `Mean` will always be selected by default, so we don't need to worry about it.
Statistics can have only two values either `Mean`  or `Median`, one is selected by default when declaring the value, we only need to care about the other one.

Reply:
But if we rely on this default, things can easily break down the line. What if:

- We change our default?
- We add more options like `Mode`?
- The user triggers a re-run of this function while `Median` is selected but the url states `Mean`?

These cases will easily break this logic, therefore it would be great to not rely on defaults, especially in a useEffect.
If we also want to add Back support, where the user can navigate backwards, point 3 will come into play here.